token/transfer-hook/cli/src/meta.rs

```diff
@@ -0,0 +1,337 @@
+use {
+    serde::{Deserialize, Serialize},
+    solana_sdk::pubkey::Pubkey,
+    spl_tlv_account_resolution::{account::ExtraAccountMeta, seeds::Seed},
+    std::{path::Path, str::FromStr},
+    strum_macros::{EnumString, IntoStaticStr},
+};
+
+#[derive(Copy, Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct Access {
+    #[serde(alias = "is_signer")]
+    is_signer: bool,
+    #[serde(alias = "is_writable")]
+    is_writable: bool,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, EnumString, IntoStaticStr, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+#[strum(serialize_all = "kebab-case")]
+enum Role {
+    Readonly,
+    Writable,
+    ReadonlySigner,
+    WritableSigner,
+}
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase", untagged)]
+enum AccessLevel {
+    Access(Access),
+    Role { role: Role },
```

Comment:
I can't remember, is there a reason to allow for both of these formats? I'm worried it'll be more confusing than helpful to allow both. I don't have a strong preference for either format

Reply:
Basically the ideas is that it's easier to provide "role" on the CLI argument `<ADDRESS>:<ROLE>` and in the file you could decide to use either one.

I guess it really isn't super necessary, and you can just use the role in the files as well. Just giving people the option. Maybe they need specifically `isSigner` and `isWritable` for some other configuration or scripting, and they can just plug that file into transfer hook configs, too.

Reply:
In that case, I'd lean towards just having the `Role`.  Let's keep it simple for the reference implementation.